### PR TITLE
Updates FAN

### PR DESCRIPTION
**Pull request by Coxswain-Navigator**
## About The Pull Request
Gives FAN a breach and lets it be worked on a meltdown.

-  At zero Qlipthoth, the air conditioning turns off in the facility. If left off for too long, players will slowly burn to death.

-  Can be turned back on by touching the abnormality.

## Why It's Good For The Game
Makes FAN less weird.

## Changelog
:cl:
tweak: FAN changed slightly
/:cl:

### CODE REVIEW

**Review by LanceSmites328 — DISMISSED** [review]

**Review by EgorDinamit — CHANGES_REQUESTED** [review]
1. Re-using sounds of other abnormalities/objects is bad.
2. You can easily instead of changing description just add it to the examination proc.
3. Do NOT use burn damage. Anything outside of brute damage is pending removal.
4. Not a request, but there are some fancy heat wave effects on Tegu you could try porting.

File: code/modules/mob/living/simple_animal/abnormality/he/fan.dm [hunk 1]
```diff
@@ -27,8 +28,12 @@
 	var/list/safe = list()
 	var/list/warning = list()
 	var/list/danger = list()
+	pet_bonus = "powers on" //saves a few lines of code by allowing funpet() to be called by attack_hand()
+	var/safework //Safe if the abnormality was melting
```
Comment (EgorDinamit) [inline code/modules/mob/living/simple_animal/abnormality/he/fan.dm:32]: If it's meant to be a boolean - set its default value to FALSE or TRUE.

File: code/modules/mob/living/simple_animal/abnormality/he/fan.dm [hunk 2]
```diff
@@ -56,3 +65,110 @@
...
+	icon_state = "fan_idle"
+	desc = "It appears to be an office fan. It looks like it's turned off."
+	HeatWave()
+
+/mob/living/simple_animal/hostile/abnormality/fan/proc/HeatWave(started)
+	if(!turned_off)
+		datum_reference.qliphoth_change(1)
+		return
+	if(!started)
+		sound_to_playing_players_on_level('sound/abnormalities/seasons/summer_idle.ogg', 150, zlevel = z)
```
Comment (EgorDinamit) [inline code/modules/mob/living/simple_animal/abnormality/he/fan.dm:114]: That's the only place the proc's argument is used. Can't you just move it to the place where you call HeatWave for the first time?

File: code/modules/mob/living/simple_animal/abnormality/he/fan.dm [hunk 3]
```diff
@@ -56,3 +65,110 @@
...
+	to_chat(owner, "<span class='warning'>You're starting to sweat.</span>")
+	if(owner.client)
+		owner.add_client_colour(/datum/client_colour/glass_colour/orange)
+
+/datum/status_effect/stacking/fanhot/add_stacks(stacks_added)
+	. = ..()
+	if(!stacks_added)
+		return
+	if(stacks > 10)
+		to_chat(owner, "<span class='warning'>IT'S TOO HOT!</span>")
```
Comment (EgorDinamit) [inline code/modules/mob/living/simple_animal/abnormality/he/fan.dm:158]: This will spam itself every 3 seconds. Reconsider.

**Review by EgorDinamit — COMMENTED** [review]
Another thing worth mentioning - absolutely disgusting implementation of the lists for the "special effect".
As it is right now - there is 3 lists of **references** to mobs. They never get cleaned up. This will chug the server with the BYOND's garbage collector.

File: code/modules/mob/living/simple_animal/abnormality/he/fan.dm [hunk 4]
```diff
@@ -56,3 +71,108 @@
...
+/mob/living/simple_animal/hostile/abnormality/fan/ZeroQliphoth(mob/living/carbon/human/user)
+	. = ..()
+	if(!turned_off)
+		TurnOff()
+
+/mob/living/simple_animal/hostile/abnormality/fan/proc/TurnOn(mob/living/carbon/human/user)
+	turned_off = FALSE
+	icon_state = "fan"
+	for(var/mob/living/carbon/human/L in GLOB.player_list)
+		if(z != L.z || L.stat >= HARD_CRIT) // on a different Z level or dead
```
Comment (EgorDinamit) [inline code/modules/mob/living/simple_animal/abnormality/he/fan.dm:100]: Why check for this? We want to remove it from everyone who was affected.

File: code/modules/mob/living/simple_animal/abnormality/he/fan.dm [hunk 5]
```diff
@@ -56,3 +71,108 @@
...
+		to_chat(petter, "<span class='nicegreen'>You hit the on switch. Aaah, that feels nice.</span>")
+		TurnOn()
+
+//Breach
+/mob/living/simple_animal/hostile/abnormality/fan/ZeroQliphoth(mob/living/carbon/human/user)
+	. = ..()
+	if(!turned_off)
+		TurnOff()
+
+/mob/living/simple_animal/hostile/abnormality/fan/proc/TurnOn(mob/living/carbon/human/user)
```
Comment (EgorDinamit) [inline code/modules/mob/living/simple_animal/abnormality/he/fan.dm:96]: Add a sound to it.

File: code/modules/mob/living/simple_animal/abnormality/he/fan.dm [hunk 6]
```diff
@@ -56,3 +71,108 @@
...
+	turned_off = FALSE
+	icon_state = "fan"
+	for(var/mob/living/carbon/human/L in GLOB.player_list)
+		if(z != L.z || L.stat >= HARD_CRIT) // on a different Z level or dead
+			continue
+		var/datum/status_effect/stacking/fanhot/V = L.has_status_effect(/datum/status_effect/stacking/fanhot)
+		if(!V)
+			continue
+		else
+			V.Destroy()
```
Comment (EgorDinamit) [inline code/modules/mob/living/simple_animal/abnormality/he/fan.dm:106]: Do NOT call destroy directly. Always use qdel()

**Review by EgorDinamit — APPROVED** [review]

**Pull request merged**